### PR TITLE
Handle error in CUDA/HIP module init and configurable max_streams

### PR DESCRIPTION
**Pull request by therault**
A misconfiguration on leconte makes none of the device capable of allocating a cuda stream.

This leads to a SEGFAULT, because we try to free things that have not been allocated.

With this patch, the devices are just removed (with a warning), but the execution proceeds.

### CODE REVIEW

**Review by abouteiller — APPROVED** [review]

File: parsec/mca/device/cuda/device_cuda_module.c [hunk 1]
```diff
@@ -363,25 +363,29 @@ parsec_cuda_module_init( int dev_id, parsec_device_module_t** module )
...
     if(-1 == len)
-        gpu_device->super.name = "";
+        gpu_device->super.name = strdup("");
     gpu_device->data_avail_epoch = 0;
 
-    gpu_device->max_exec_streams = PARSEC_MAX_STREAMS;
+    gpu_device->max_exec_streams = 0; // We will increment this as streams are succesfully initialized
     gpu_device->exec_stream =
-        (parsec_gpu_exec_stream_t**)malloc(gpu_device->max_exec_streams * sizeof(parsec_gpu_exec_stream_t*));
+        (parsec_gpu_exec_stream_t**)malloc(PARSEC_MAX_STREAMS * sizeof(parsec_gpu_exec_stream_t*));
```
Comment (bosilca) [inline parsec/mca/device/cuda/device_cuda_module.c:371]: Why are we removing the capability to dynamically define the number of execution streams per device ?

Comment (bosilca) [reply parsec/mca/device/cuda/device_cuda_module.c:371]: This was part of the infrastructure to allow the number of exec_streams to become an MCA param. I would prefer to complete the work than remove it.

Comment (abouteiller) [reply parsec/mca/device/cuda/device_cuda_module.c]: I updated the PR to add configurable max_streams

**Review by abouteiller — APPROVED** [review]

File: parsec/mca/device/cuda/device_cuda_component.c [hunk 2]
```diff
@@ -189,9 +195,6 @@ static int device_cuda_component_register(void)
     (void)parsec_mca_param_reg_int_name("device_cuda", "max_number_of_ejected_data",
                                         "Sets up the maximum number of blocks that can be ejected from GPU memory",
                                         false, false, MAX_PARAM_COUNT, &parsec_gpu_d2h_max_flows);
-    (void)parsec_mca_param_reg_int_name("device_cuda", "sort_pending_tasks",
```
Comment (bosilca) [inline parsec/mca/device/cuda/device_cuda_component.c:192]: Move back to original order.

**Review by bosilca — APPROVED** [review]

**Pull request merged**